### ISSUE TITLE
Pitest fails on passing test

### ISSUE THREAD

**Issue by bnvinay92** [koral--/gradle-pitest-plugin#4]
I'm using RxJava2 in my test, and this line:
 `AndroidSchedulers.mainThread()` 
causes pit to fail even though the test passes. I have `testOptions.unitTests.returnDefaultValues = true`, which stops getMainLooper() (used by `AndroidSchedulers.mainThread()` internally) from throwing an exception, which might have something to do with it.

**Comment by koral--** [comment]
Could you enable verbose logging (`verbose = true` in `pitest` stanza) and post log here? 
Or even better post the minimal unit test which reproduces this issue?

**Comment by bnvinay92** [comment]
This repo reproduces this error: https://github.com/bnvinay92/reproduce-pitest-error

JUnit4 fails with the same error if i set `testOptions.unitTests.returnDefaultValues = false`.
Setting default values to `true` causes pitest to fail with the error.

Stacktrace:

```
8:24:38 PM PIT >> FINE : Running report with ReportOptions [targetClasses=[^com\.github\.pitesterror\..*$], excludedMethods=[], excludedClasses=[^.*.*AutoValue_.*$], codePaths=[/Users/Vinay/AndroidStudioProjects/PitestError/app/build/intermediates/classes/debug], reportDir=/Users/Vinay/AndroidStudioProjects/PitestError/app/build/reports/pitest/debug, historyInputLocation=null, historyOutputLocation=null, sourceDirs=[/Users/Vinay/AndroidStudioProjects/PitestError/app/src/main/java], classPathElements=[/Users/Vinay/.gradle/caches/modules-2/files-2.1/org.pitest/pitest/1.1.11/a382a5b55bb2bdcd4210783709470ff0076cee14/pitest-1.1.11.jar, /Users/Vinay/Library/Android/sdk/extras/android/m2repository/com/android/support/appcompat-v7/25.3.1/appcompat-v7-25.3.1.aar, /Users/Vinay/Library/Android/sdk/extras/m2repository/com/android/support/constraint/constraint-layout/1.0.2/constraint-layout-1.0.2.aar, /Users/Vinay/.gradle/caches/modules-2/files-2.1/io.reactivex.rxjava2/rxandroid/2.0.1/c192035f52624f35143cadfc9d5832cd621319eb/rxandroid-2.0.1.aar, /Users/Vinay/.gradle/caches/modules-2/files-2.1/io.reactivex.rxjava2/rxjava/2.1.0/2fdf84dedcaaeabb9d70cde9dbb8aad4eccb80a1/rxjava-2.1.0.jar, /Users/Vinay/Library/Android/sdk/extras/android/m2repository/com/android/support/support-annotations/25.3.1/support-annotations-25.3.1.jar, /Users/Vinay/Library/Android/sdk/extras/android/m2repository/com/android/support/support-v4/25.3.1/support-v4-25.3.1.aar, /Users/Vinay/Library/Android/sdk/extras/android/m2repository/com/android/support/support-vector-drawable/25.3.1/support-vector-drawable-25.3.1.aar, /Users/Vinay/Library/Android/sdk/extras/android/m2repository/com/android/support/animated-vector-drawable/25.3.1/animated-vector-drawable-25.3.1.aar, /Users/Vinay/Library/Android/sdk/extras/m2repository/com/android/support/constraint/constraint-layout-solver/1.0.2/constraint-layout-solver-1.0.2.jar, /Users/Vinay/Library/Android/sdk/extras/android/m2repository/com/android/support/support-compat/25.3.1/support-compat-25.3.1.aar, /Users/Vinay/Libr                                                                          ary/Android/sdk/extras/android/m2repository/com/android/support/support-media-compat/25.3.1/support-media-compat-25.3.1.aar, /Users/Vinay/Library/Android/sdk/extras/android/m2repository/com/android/support/support-core-utils/25.3.1/support-core-utils-25.3.1.aar, /Users/Vinay/Library/Android/sdk/extras/android/m2repository/com/android/support/support-core-ui/25.3.1/support-core-ui-25.3.1.aar, /Users/Vinay/Library/Android/sdk/extras/android/m2repository/com/android/support/support-fragment/25.3.1/support-fragment-25.3.1.aar, /Users/Vinay/.gradle/caches/modules-2/files-2.1/org.reactivestreams/reactive-streams/1.0.0/14b8c877d98005ba3941c9257cfe09f6ed0e0d74/reactive-streams-1.0.0.jar, /Users/Vinay/.gradle/caches/modules-2/files-2.1/junit/junit/4.12/2973d150c0dc1fefe998f834810d68f278ea58ec/junit-4.12.jar, /Users/Vinay/.gradle/caches/modules-2/files-2.1/org.hamcrest/hamcrest-core/1.3/42a25dc3219429f0e5d060061f71acb49bf010a0/hamcrest-core-1.3.jar, /Users/Vinay/AndroidStudioProjects/PitestError/app/build/intermediates/sourceFolderJavaResources/debug, /Users/Vinay/AndroidStudioProjects/PitestError/app/build/intermediates/sourceFolderJavaResources/test/debug, /Users/Vinay/.android/build-cache/3fcaca07f47f57c52b107d87948f5f88afc1fea2/output/jars/classes.jar, /Users/Vinay/.android/build-cache/460d707dc826f1b8788374e17ea157fdfac43f9e/output/jars/classes.jar, /Users/Vinay/.android/build-cache/ec5d4cd7d605e5ba046a6aa57a2f6e493c5d9f8b/output/jars/classes.jar, /Users/Vinay/.android/build-cache/a501270de1df8b15b4b7d7b4b0da954c6bcf77ed/output/jars/classes.jar, /Users/Vinay/.android/build-cache/97e7706b9c93b5cf621d775aba863ad903444a27/output/jars/classes.jar, /Users/Vinay/.android/build-cache/0484e9aabd44ebb3b478ccd7646a20c3aa74eec1/output/jars/classes.jar, /Users/Vinay/.android/build-cache/397532c8d3a6df988e522fd0d550a600c942097b/output/jars/classes.jar, /Users/Vinay/.android/build-cache/e5d51b35693ca2b2f24b9c8e72715dcacbc879b2/output/jars/classes.jar, /Users/Vinay/.android/build-cache/20043adc95aba96e39094afd29e839236c6b7f28/output/jars/classes.jar, /Users/Vinay/.android/build-cache/33c54378096730c9bf5ab374a7130dab678d31fc/output/jars/classes.jar, /Users/Vinay/.android/build-cache/b845a8fb6eda5e13761f7c8e0561b9dac61b6c62/output/jars/classes.jar, /Users/Vinay/AndroidStudioProjects/PitestError/app/build/intermediates/classes/debug, /Users/Vinay/AndroidStudioProjects/PitestError/app/build/intermediates/classes/test/debug, /Users/Vinay/.gradle/caches/modules-2/files-2.1/org.robolectric/android-all/7.1.0_r7-robolectric-0/c6478c2c63230d9480023ea5d90180c1e4f3b893/android-all-7.1.0_r7-robolectric-0.jar], mutators=[], dependencyAnalysisMaxDistance=-1, mutateStaticInitializers=false, jvmArgs=[], numberOfThreads=9, timeoutFactor=1.25, timeoutConstant=4000, targetTests=[], loggingClasses=[], maxMutationsPerClass=0, verbose=true, failWhenNoMutations=true, outputs=[HTML], groupConfig=TestGroupConfig [excludedGroups=[], includedGroups=[]], mutationUnitSize=0, shouldCreateTimestampedReports=false, detectInlinedCode=false, exportLineCoverage=false, mutationThreshold=0, coverageThreshold=0, mutationEngine=gregor, javaExecutable=null, includeLaunchClasspath=false, properties={}, maxSurvivors=-1, excludedRunners=[]]
8:24:38 PM PIT >> FINE : System class path is /Users/Vinay/.gradle/caches/modules-2/files-2.1/org.pitest/pitest-command-line/1.1.11/8885860b0d0416230e27ed2ed9a065e0c17dc504/pitest-command-line-1.1.11.jar:/Users/Vinay/.gradle/caches/modules-2/files-2.1/org.robolectric/android-all/7.1.0_r7-robolectric-0/c6478c2c63230d9480023ea5d90180c1e4f3b893/android-all-7.1.0_r7-robolectric-0.jar:/Users/Vinay/.gradle/caches/modules-2/files-2.1/org.pitest/pitest/1.1.11/a382a5b55bb2bdcd4210783709470ff0076cee14/pitest-1.1.11.jar:/Users/Vinay/.gradle/caches/modules-2/files-2.1/junit/junit/4.11/4e031bb61df09069aeb2bffb4019e7a5034a4ee0/junit-4.11.jar:/Users/Vinay/.gradle/caches/modules-2/files-2.1/org.hamcrest/hamcrest-core/1.3/42a25dc3219429f0e5d060061f71acb49bf010a0/hamcrest-core-1.3.jar
8:24:38 PM PIT >> FINE : Maximum available memory is 3641 mb
8:24:39 PM PIT >> INFO : MINION : Picked up JAVA_TOOL_OPTIONS: -Djava.awt.headless=true
objc[53950]: Class JavaLaunchHelper is implemented in both /Library/Java/JavaVirtualMachines/jdk1.8.0_51.jdk/Contents/Home/jre/bin/java and /Library/Java/JavaVirtualMachines/jdk1.8.0_51.jdk/Contents/Hom
8:24:39 PM PIT >> FINE : MINION : Installing PIT agent

8:24:39 PM PIT >> INFO : MINION : e/jre/lib/libinstrument.dylib. One of the two will be used. Which one is undefined.

8:24:39 PM PIT >> INFO : Sending 2 test classes to minion
8:24:39 PM PIT >> INFO : Sent tests to minion
8:24:39 PM PIT >> INFO : MINION : 8:24:39 PM PIT >> INFO : Checking environment

8:24:39 PM PIT >> INFO : MINION : 8:24:39 PM PIT >> FINE : Expecting 2 tests classes from parent

8:24:39 PM PIT >> INFO : MINION : 8:24:39 PM PIT >> FINE : Tests classes received

8:24:39 PM PIT >> INFO : MINION : 8:24:39 PM PIT >> INFO : Found  2 tests

8:24:39 PM PIT >> INFO : MINION : 8:24:39 PM PIT >> INFO : Dependency analysis reduced number of potential tests by 0

8:24:39 PM PIT >> INFO : MINION : 8:24:39 PM PIT >> INFO : 2 tests received

8:24:39 PM PIT >> INFO : MINION : 8:24:39 PM PIT >> FINE : Running 2 units

8:24:39 PM PIT >> INFO : MINION : 8:24:39 PM PIT >> FINE : Gathering coverage for test Description [testClass=com.github.pitesterror.ExampleUnitTest, name=addition_isCorrect(com.github.pitesterror.ExampleUnitTest)]

8:24:39 PM PIT >> INFO : MINION : 8:24:39 PM PIT >> FINE : Gathering coverage for test Description [testClass=com.github.pitesterror.PitestErrorTest, name=returnsOne(com.github.pitesterror.PitestErrorTest)]

8:24:39 PM PIT >> FINE : MINION : FAIL Description [testClass=com.github.pitesterror.PitestErrorTest, name=returnsOne(com.github.pitesterror.PitestErrorTest)] -> java.lang.ExceptionInInitializerError

8:24:39 PM PIT >> SEVERE : Description [testClass=com.github.pitesterror.PitestErrorTest, name=returnsOne(com.github.pitesterror.PitestErrorTest)] did not pass without mutation.
8:24:39 PM PIT >> FINE : Coverage generator Minion exited ok
8:24:39 PM PIT >> INFO : Calculated coverage in 0 seconds.
Exception in thread "main" org.pitest.help.PitHelpError: All tests did not pass without mutation when calculating line coverage. Mutation testing requires a green suite.
See http://pitest.org for more details.
        at org.pitest.coverage.execute.DefaultCoverageGenerator.verifyBuildSuitableForMutationTesting(DefaultCoverageGenerator.java:109)
        at org.pitest.coverage.execute.DefaultCoverageGenerator.calculateCoverage(DefaultCoverageGenerator.java:94)
        at org.pitest.coverage.execute.DefaultCoverageGenerator.calculateCoverage(DefaultCoverageGenerator.java:49)
        at org.pitest.mutationtest.tooling.MutationCoverage.runReport(MutationCoverage.java:112)
        at org.pitest.mutationtest.tooling.EntryPoint.execute(EntryPoint.java:103)
        at org.pitest.mutationtest.tooling.EntryPoint.execute(EntryPoint.java:45)
        at org.pitest.mutationtest.commandline.MutationCoverageReport.runReport(MutationCoverageReport.java:87)
        at org.pitest.mutationtest.commandline.MutationCoverageReport.main(MutationCoverageReport.java:45)
:app:pitestDebug FAILED

```

**Comment by koral--** [comment]
Thanks, will check it.

**Comment by koral--** [comment]
Tests calling methods from Android jar does not work with pitest. It seems that it falls through the last case there: http://pitest.org/faq/#my-tests-normally-run-green-but-pit-says-the-suite-isnt-green

To solve it you can add mockable android jar with default values to the test classpath - [example](https://github.com/koral--/reproduce-pitest-error/commit/214e98d8c3e48d862df91a598f06db59844fea44)
You can additionally extract hardcoded number `25` so it will be easier to maintain if API level changes in the future.

It seems to be a quite common case to use RxJava. I'll add a note about that to README.

**Comment by koral--** [comment]
I've decided to include this feature in the plugin. So you only need to set `testOptions.unitTests.returnDefaultValues = true` in such cases.
Change Available in version 0.0.7.

**Comment by bnvinay92** [comment]
Thanks!
